### DISCUSSION TITLE
How do I access my theme in component styles?

### DISCUSSION THREAD

**Discussion by JohannesOlsson8** [microsoft/fluentui#20910] [Q&A]
Hi,

I've been searching for an answer all morning without result, hopefully you guys can help me. 

My project is wrapped in a <ThemeProvider> and my themes are working well. 

The problem is that I can't access it in any ...styles.ts file. I've tried using the useTheme hook but it only returns some default theme for Fluent. 

I've also tried this (deprecated). 

`import { makeStyles } from "@fluentui/react";

export const useFooStyles = makeStyles((theme) => ({
  root: {
    width: "100px",
    height: "100px",
    background: theme.palette.themePrimary,
  },
}));
`

What am I missing?

**Comment by khmakoto** [comment]
Hi @JohannesOlsson8, which version of the product are you using?

**Reply by JohannesOlsson8** [reply]
Hi @khmakoto!  

@fluentui/react: 8.46.1
